src/geojson-layer.ts

```diff
@@ -98,8 +107,26 @@ export default class GeoJSONLayer extends React.Component<Props, {}> {
       },
       before
     );
+
+    this.mapLayerMouseHandlers(type)
   };
 
+  private mapLayerMouseHandlers = (type: string) => {
+    const { map } = this.context;
+
+    const layerId = `${this.id}-${type}`;
```

Comment:
Could you create a function `buildLayerId` and use it in line 88 too?

Reply:
done